frontend/src/bundles/common/hooks/use-form-error/use-form-error.hook.ts

```diff
@@ -0,0 +1,39 @@
+import { type UseFormSetError } from 'react-hook-form';
+import {
+    type UserSignInRequestDto,
+    type UserSignUpRequestDto,
+} from 'shared/build';
+
+import { DataStatus } from '~/bundles/common/enums/data-status.enum';
+import { useEffect } from '~/bundles/common/hooks/hooks';
+import { useAppSelector } from '~/bundles/common/hooks/use-app-selector/use-app-selector.hook';
+
+type UserAuthDto = UserSignUpRequestDto | UserSignInRequestDto;
+
+type UseFormErrorPayload = {
+    setError: UseFormSetError<UserAuthDto>;
+};
+
+type ReturnValue = {
+    dataStatus: DataStatus;
+};
+
+const useFormError = ({ setError }: UseFormErrorPayload): ReturnValue => {
+    const { error, dataStatus } = useAppSelector((state) => state.auth);
+
+    useEffect(() => {
+        if (error && dataStatus === DataStatus.REJECTED) {
```

Comment:
Is there a case where it's error and dataStatus is not rejected?

Reply:
yes, data status check is redundant)